serf/snapshot.go

```diff
@@ -311,6 +313,18 @@ func (s *Snapshotter) processQuery(q *Query) {
 func (s *Snapshotter) tryAppend(l string) {
 	if err := s.appendLine(l); err != nil {
 		s.logger.Printf("[ERR] serf: Failed to update snapshot: %v", err)
+		now := time.Now()
+		if now.Sub(s.lastAttemptedCompaction) > snapshotErrorRecoveryInterval {
+			s.lastAttemptedCompaction = now
+			wrErr := err
+			s.logger.Printf("[ERR] serf: Attempting compaction to recover from error...")
+			err = s.compact()
+			if err != nil {
+				s.logger.Printf("[ERR] serf: Failed compaction due to error %v, will reattempt in %v", err, snapshotErrorRecoveryInterval)
```

Comment:
Errors in Go often have multiple parts, so don't inline very well, I'd change this to:

`[ERR] serf: Compaction failed, will reattempt after %v: %v", snapshotErrorRecoveryInterval, err`

and drop the original error since it was already printed up on line 315:

`[INFO] serf: Finished compaction, successfully recovered from error state`